libraries/apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/ApolloCall.kt

```diff
@@ -93,16 +106,36 @@ class ApolloCall<D : Operation.Data> internal constructor(
         .sendDocument(sendDocument)
         .enableAutoPersistedQueries(enableAutoPersistedQueries)
         .canBeBatched(canBeBatched)
+        .throwOnException(throwOnException)
         .build()
     return apolloClient.executeAsFlow(request)
   }
 
   /**
-   * A shorthand for `toFlow().single()`.
-   * Use this for queries and mutation to get a single [ApolloResponse] from the network or the cache.
-   * For subscriptions, you usually want to use [toFlow] instead to listen to all values.
+   * Retrieve a single [ApolloResponse] from this [ApolloCall], ignoring any cache misses or network errors.
+   *
+   * Use this for queries and mutations to get a single value from the network or the cache.
+   * For subscriptions or operations using `@defer`, you usually want to use [toFlow] instead to listen to all values.
+   *
+   * @throws ApolloException if the call returns zero or multiple valid GraphQL responses.
```

Comment:
Trying to wrap my head around this:

* throwOnException = false
  * foldFetchException = false 
    * cache first + cache error + network success -> returns the network response, no need to have info about the cache miss
    * cache first + cache error + network error -> returns response.exception = ApolloCompositeException() with both exceptions
  * foldFetchException = true
    * cache first + cache error + network success -> returns the network response, cache miss is in response.cacheInfo.cacheMissException
    * cache first + cache error + network error -> returns response.exception = ApolloCompositeException() with both 
* throwOnException = true
  * foldFetchException = false 
    * cache first + cache error + network success -> returns the network response, no need to have info about the cache miss
    * cache first + cache error + network error -> throws ApolloCompositeException() with both exceptions
* throwOnException = true
  * foldFetchException = true 
    * cache first + cache error + network success -> returns the network response, cache miss is in response.cacheInfo.cacheMissException
    * cache first + cache error + network error -> throws ApolloCompositeException() with both exceptions

Did I get this right? 

I'm starting to question the need for a separate `foldFetchException`. We can keep it internally of course but given this is only for backward compat, there's no real need to decouple both parameters?

Reply:
This looks 💯 correct to me. 

And yes, good point, as a user migrating to v4 this would probably be more confusing than needed!. A single flag to replicate the v3 behavior is probably better. Now the difficult question: how to name the flag? 😅

`foldAndThrowExceptions`?

Reply:
Something else I just realized is one is in `apollo-runtime` and the other in `apollo-normalized-cache` (fun times!)

Maybe make it explicit about v3 compat? 

* apollo-runtime: `throwOnExceptions()`?
* apollo-normalized-cache: `useV3ExceptionHandling()` ?  (calls `throwOnExceptions()`)

I'm not 100% sold TBH. I think we'll need to check the whole story in Confetti/tutorial/other repos to double check the ergonomics later on.

Reply:
What about `useV3ExceptionHandling()` only?

Reply:
In both artifacts? Feels dangerous because it's very easy to import the wrong one.

Reply:
No sorry I meant only in apollo-runtime, and it could control both behaviors. Or am I missing something?

Reply:
That would force exposing the parameter outside of `apollo-runtime` (for `apollo-normalized-cache` to be able to read it). Works for me with some `@ApolloInternal` annotation?